src/NHibernate/Tool/hbm2ddl/SchemaMetadataUpdater.cs

```diff
@@ -56,27 +57,45 @@ public static void QuoteTableAndColumns(Configuration configuration, Dialect.Dia
 		{
 			foreach (var cm in configuration.ClassMappings)
 			{
-				QuoteTable(cm.Table, dialect);
+				QuoteTable(cm.Table,cm.Key?.ColumnIterator?.OfType<Column>(), dialect);
 			}
+
 			foreach (var cm in configuration.CollectionMappings)
 			{
-				QuoteTable(cm.Table, dialect);
+				QuoteTable(cm.Table, cm.Key?.ColumnIterator?.OfType<Column>(), dialect);
```

Comment:
What about `cm.CollectionTable`? Can you check if we have those columne there. Maybe whole fix should be as simple as:
```suggestion
				QuoteTable(cm.Table, dialect);
				QuoteTable(cm.CollectionTable, dialect);
```

Reply:
I've checked. It doesn't work.

Reply:
Yeah I see different column instances in `cm,Keys` and inside `cm.CollectionTable`. And it seems reusing existing table columns would fix this issue without any additional changes. But not sure if it's a good idea. Maybe not.